### ISSUE TITLE
Can this method work on APK's that are already compiled?

### ISSUE THREAD

**Issue by NeoToriyama** [tomthecarrot/arcore-for-all#9]
Basically, the title is the issue. Some people on YouTube have released test APK's for ARCore but not the project files. Obviously, they aren't supported because they don't use the modified .arr file. If we have an APK file, how can it be used in combination with this method? If that's possible.

(I'm using the Samsung Galaxy S8+ btw)

**Comment by tomthecarrot** [comment]
You can use [apktool](https://ibotpeaches.github.io/Apktool/documentation/) to decompile the APK, then follow the Android build instructions in this repo's [README](https://github.com/tomthecarrot/arcore-for-all/blob/master/README.md) to replace the library. Then rebuild the APK using apktool and sign it using [apksigner](https://developer.android.com/studio/command-line/apksigner.html) before installing it to your Android device via adb.

**Comment by NeoToriyama** [comment]
Okay, so I learned how to decompile and recompile the APK, but the only instructions I see are for modifying the aar library. Here's the folder structure of the decompiled APK: http://imgur.com/a/w9uRI

How would I go about modifying it with your method from this point forward? Thanks for all the help so far.

**Comment by tomthecarrot** [comment]
The AAR file you need to modify (or replace with the one in this repo) should be inside the `libs` directory.

**Comment by NeoToriyama** [comment]
Okay, so here's the folder structure within the libs directory: http://imgur.com/a/w9uRI
The file doesn't seem to exist. Could it be hidden?

**Comment by tomthecarrot** [comment]
Ah I see. This is a compiled Unity project. Unity compiles multiple project libraries into shared object files. You might be able to decompile the shared object containing the AAR (assuming that it's still in that format), replace the AAR, and recompile. It would be simplest to modify the original Unity project, if you have access to it.

**Comment by NeoToriyama** [comment]
No, it was uploaded online so I don't have access. I also don't have Unity installed either. Is there possibly a way to modify the ARCore service APK provided by Google to support unsupported devices. Or can it only be currently accomplished on the actual app?

**Comment by tomthecarrot** [comment]
That's an interesting thought. It would make this modification far more universal. I'll look into it.

**Comment by NeoToriyama** [comment]
It's definitely something I've been contemplating lately. It could be like a hook service that runs on the phone so when you load up an APK using ARCore, it would spoof your device as compatible or disable the compatibility check. Although, something on this level might require root access. Thanks for all the help. I'm installing Unity now in hopes of using the method you mentioned above. I'll let you know how it goes.

**Comment by NeoToriyama** [comment]
Here's the link to the APK if you want to take a crack at it.
http://208107.selcdn.ru/TestVR_OPEN/TestVR_RC3.apk

**Comment by tomthecarrot** [comment]
Cool! Thank you. I built an early version of a [modified ARCore Service](tomthecarrot.com/projects/arcore-for-all/arcore-preview-modified.apk), which seems to be functional without per-app modifications, all on my "unsupported" device.

Please try it on your device and let me know if it works. Thanks!

**Comment by NeoToriyama** [comment]
Thank you,  but the link appears to be down.

**Comment by tomthecarrot** [comment]
Sorry about that. I just updated the comment's URL.

**Comment by NeoToriyama** [comment]
No worries,  I installed it but no luck yet. I'm gonna restart my phone and see if it works then.

**Comment by NeoToriyama** [comment]
Okay, so the unmodified HelloAR Java apk works without the modified aar file, but the other app I downloaded still displays the "unsupported" message. So far this is good as it seems the modified Tango Core apk does it's job for the most part. Hopefully Google will release new apps so we can determine if it only works with their apps or if it's dependant on which SDK the app is built under. Thanks again.

**Comment by tomthecarrot** [comment]
@NeoToriyama That's good to hear! Does it actually show a camera with full ARCore tracking?

**Comment by NeoToriyama** [comment]
Yes it does and the tracking works very well. I'm using the Samsung Galaxy S8+.

**Comment by tomthecarrot** [comment]
Thanks! Will close this once we complete testing and build instructions for the shared library.

**Comment by NeoToriyama** [comment]
Of course and no thank you for all the work. It's greatly appreciated!

**Comment by NeoToriyama** [comment]
Just confirmed that the tracking is much better on the unmodified Java APK. I don't know how it's possible, but after several tests back and forth I'm fairly certain. This could due to fact that it's a Java build. It also loads much faster than the modified HelloAR apk.

**Comment by tomthecarrot** [comment]
Good to know for testing the issue on other devices, thank you!

**Comment by NeoToriyama** [comment]
You're welcome!

**Comment by Indashio** [comment]
Not working on Xperia XZ Premium
